src/jobs/deploy-eb.yml

```diff
@@ -1,9 +1,88 @@
 description: >
-  This job deploys an application to aws elasticbeanstalk.
+  This job deploys an application to aws elasticbeanstalk. Deploy-Eb requires publish-docker
 
 executor: node/default
 
+parameters:
+  # Required Parameters
+  application-name:
+    type: string
+    description: |
+      The name of the elastic beanstalk instance that will be updated in this job
+  environment-name:
+    type: string
+    description: |
+      The name of the elastic beanstalk instance that was given during eb init
+  repo-name:
+    type: string
+    description: |
+      The Github repository name of the elasticbeanstalk instance
+      This will be used for tagging
+  version:
+    type: string
+    description: |
+      The semantic version of the eb instance
+  # Not Required Parameters
+  aws-secret-access-key-id:
+    type: env_var_type
+    default: AWS_SECRET_ACCESS_KEY_ID
+    description: |
+      The secret key for the aws user
+  aws-access-key-id:
+    type: env_var_type
+    default: AWS_ACCESS_KEY_ID
+    description: |
+      The access key for the aws user
+  aws-region:
+    type: env_var_type
+    default: AWS_REGION
+    description: |
+      The region where the elasticbeanstalk instance is located
+  is-docker:
+    type: boolean
+    default: true
+    description: |
+      The platform of the elastic beanstalk instance will be deployed with docker
+  is-node:
+    type: boolean
+    default: false
+    description: |
+      The platform of the elastic beanstalk instance will be deployed with nodejs
+
 steps:
```

Comment:
we persisted build information from the previous job but never attached it in this job.